[pt][quant] Convert conv_prepack to conv2d_prepack and conv_unpack to conv2d_unpack

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29529 [pt][quant] Convert conv_prepack to conv2d_prepack and conv_unpack to conv2d_unpack**

We would like to replace `conv_prepack` with `conv2d_prepack` and  `conv_unpack` with `conv2d_unpack`.

This makes the naming consistent between 2D and 3D conv:
```
torch.ops.quantized.conv2d_prepack
torch.ops.quantized.conv2d_unpack
torch.ops.quantized.conv2d
torch.ops.quantized.conv3d_prepack
torch.ops.quantized.conv3d_unpack
torch.ops.quantized.conv3d
```

We should do this earlier rather than later when we have more users for the quantized conv2d ops, for better engineering.

The replacement bash command is as the follows:
```
find ./ -type f -exec sed -i -e 's/quantized::conv_prepack/quantized::conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/quantized::conv_unpack/quantized::conv2d_unpack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv_prepack/torch.ops.quantized.conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv_unpack/torch.ops.quantized.conv2d_unpack/g' {} \;
```

Differential Revision: [D18421079](https://our.internmc.facebook.com/intern/diff/D18421079/)

jianyuh committed:
[pt][quant] Convert conv_prepack to conv2d_prepack and conv_unpack to conv2d_unpack

We would like to replace `conv_prepack` with `conv2d_prepack` and  `conv_unpack` with `conv2d_unpack`.

This makes the naming consistent between 2D and 3D conv:
```
torch.ops.quantized.conv2d_prepack
torch.ops.quantized.conv2d_unpack
torch.ops.quantized.conv2d
torch.ops.quantized.conv3d_prepack
torch.ops.quantized.conv3d_unpack
torch.ops.quantized.conv3d
```

We should do this earlier rather than later when we have more users for the quantized conv2d ops, for better engineering.

The replacement bash command is as the follows:
```
find ./ -type f -exec sed -i -e 's/quantized::conv2d_prepack/quantized::conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/quantized::conv2d_unpack/quantized::conv2d_unpack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_prepack/torch.ops.quantized.conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_unpack/torch.ops.quantized.conv2d_unpack/g' {} \;
```

Differential Revision: [D18421079](https://our.internmc.facebook.com/intern/diff/D18421079/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][quant] Convert conv_prepack to conv2d_prepack and conv_unpack to conv2d_unpack"


We would like to replace `conv_prepack` with `conv2d_prepack` and  `conv_unpack` with `conv2d_unpack`.

This makes the naming consistent between 2D and 3D conv:
```
torch.ops.quantized.conv2d_prepack
torch.ops.quantized.conv2d_unpack
torch.ops.quantized.conv2d
torch.ops.quantized.conv3d_prepack
torch.ops.quantized.conv3d_unpack
torch.ops.quantized.conv3d
```

We should do this earlier rather than later when we have more users for the quantized conv2d ops, for better engineering.

The replacement bash command is as the follows:
```
find ./ -type f -exec sed -i -e 's/quantized::conv2d_prepack/quantized::conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/quantized::conv2d_unpack/quantized::conv2d_unpack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_prepack/torch.ops.quantized.conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_unpack/torch.ops.quantized.conv2d_unpack/g' {} \;
```

Differential Revision: [D18421079](https://our.internmc.facebook.com/intern/diff/D18421079/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt][quant] Convert conv_prepack to conv2d_prepack and conv_unpack to conv2d_unpack"


We would like to replace `conv_prepack` with `conv2d_prepack` and  `conv_unpack` with `conv2d_unpack`.

This makes the naming consistent between 2D and 3D conv:
```
torch.ops.quantized.conv2d_prepack
torch.ops.quantized.conv2d_unpack
torch.ops.quantized.conv2d
torch.ops.quantized.conv3d_prepack
torch.ops.quantized.conv3d_unpack
torch.ops.quantized.conv3d
```

We should do this earlier rather than later when we have more users for the quantized conv2d ops, for better engineering.

The replacement bash command is as the follows:
```
find ./ -type f -exec sed -i -e 's/quantized::conv2d_prepack/quantized::conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/quantized::conv2d_unpack/quantized::conv2d_unpack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_prepack/torch.ops.quantized.conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_unpack/torch.ops.quantized.conv2d_unpack/g' {} \;
```

Differential Revision: [D18421079](https://our.internmc.facebook.com/intern/diff/D18421079/)

[ghstack-poisoned]

dzhulgakov commented:
Let's bind implementations to both new (conv2d_prepack) and old (conv_prepack) APIs for now as there might be existing JIT models that have conv_prepack calls serialized there.

The alternative would be to do some "read adapter" in JIT deserialization but it seems not that different from just binding the API twice. The downside is that it'd be still exposed to users. Probably we need to add some mechanism to mark ops as "deprecated" and producing corresponding Python bindings for warnings in this case.

cc @houseroad @smessmer for an example of BC change and op deprecation

dzhulgakov commented on aten/src/ATen/native/quantized/cpu/qconv_unpack.cpp:
```diff
@@ -146,6 +146,10 @@ static auto registry =
             " -> (Tensor unpacked_weights, Tensor? B_origin)",
             c10::RegisterOperators::options().kernel<QConvUnpackWeightsInt8<2>>(
                 TensorTypeId::CPUTensorId))
+        .op("quantized::conv2d_unpack(Tensor packed_weights)"
```
can you add a comment on why we bind both and that one is deprecated? same for the _pack

jianyuh committed:
Update on "[pt][quant] Convert conv_prepack to conv2d_prepack and conv_unpack to conv2d_unpack"


We would like to replace `conv_prepack` with `conv2d_prepack` and  `conv_unpack` with `conv2d_unpack`.

This makes the naming consistent between 2D and 3D conv:
```
torch.ops.quantized.conv2d_prepack
torch.ops.quantized.conv2d_unpack
torch.ops.quantized.conv2d
torch.ops.quantized.conv3d_prepack
torch.ops.quantized.conv3d_unpack
torch.ops.quantized.conv3d
```

We should do this earlier rather than later when we have more users for the quantized conv2d ops, for better engineering.

The replacement bash command is as the follows:
```
find ./ -type f -exec sed -i -e 's/quantized::conv2d_prepack/quantized::conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/quantized::conv2d_unpack/quantized::conv2d_unpack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_prepack/torch.ops.quantized.conv2d_prepack/g' {} \;
find ./ -type f -exec sed -i -e 's/torch.ops.quantized.conv2d_unpack/torch.ops.quantized.conv2d_unpack/g' {} \;
```

Differential Revision: [D18421079](https://our.internmc.facebook.com/intern/diff/D18421079/)

[ghstack-poisoned]

z-a-f reviewed (approved):
LGTM
